// vue-route-generator/lib/template/routes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const prettier = require("prettier");
function createChildrenRoute(children) {
    return `,
    children: [${children.map(createRoute).join(',')}]`;
}
function createRoute(meta) {
    const children = !meta.children ? '' : createChildrenRoute(meta.children);
    const routeMeta = !meta.routeMeta
        ? ''
        : ',\nmeta: ' + JSON.stringify(meta.routeMeta, null, 2);
    return `
  {
    name: '${meta.name}',
    path: '${meta.path}',
    component: ${meta.specifier}${routeMeta}${children}
  }`;
}
function createImport(meta, dynamic) {
    const code = dynamic
        ? `const ${meta.specifier} = () => import(/* webpackChunkName: "${meta.name}" */ '${meta.component}')`
        : `import ${meta.specifier} from '${meta.component}'`;
    return meta.children
        ? [code]
            .concat(meta.children.map(child => createImport(child, dynamic)))
            .join('\n')
        : code;
}
function createRoutes(meta, dynamic) {
    const imports = meta.map(m => createImport(m, dynamic)).join('\n');
    const code = meta.map(createRoute).join(',');
    return prettier.format(`${imports}\n\nexport default [${code}]`, {
        parser: 'babylon',
        semi: false,
        singleQuote: true
    });
}
exports.createRoutes = createRoutes;
